normandy/recipes/management/commands/sync_remote_settings.py

```diff
@@ -79,3 +81,7 @@ def handle(self, *args, dry_run=False, **options):
             self.stdout.write(f" * {name!r} (id={r.id!r})")
             if not dry_run:
                 remote_settings.unpublish(r)
+
+        # When running with dry-run, we return non-zero if some differences exist.
+        if dry_run and (to_publish or to_update or to_unpublish):
+            sys.exit(1)
```

Comment:
I would use an exit code other than 1, so we can distinguish differences existing from the script exiting due to an unhandled exception.

Reply:
I would raise an exception instead. It's a lot less subtle and it automatically gives us non-zero exit codes for the DevOps purposes. That'd feel more pythonic. 

If raising an exception triggers something to Sentry during this testing then that's unfortunate but it's easily solved with the `ignore_exceptions` tuple config. (I think that's what it's called).